### DISCUSSION TITLE
`settings.default` are not getting reflected

### DISCUSSION THREAD

**Discussion by conscious-puppet** [srid/haskell-flake#176] [Q&A] [answered]
Tried to apply default settings to all packages, but the settings are not getting reflected. 

I still see `checkPhase` and `haddockPhase` for packages.

In the following instance, since `jailbreak = true` setting is not applied, `nix build` fails.
The build works fine when I target individual package settings, i.e, `settings.hashable.jailbreak = true`

settings applied:
```
settings.default = { ... }:
  {
    check = false;
    haddock = false;
    jailbreak = true;
    libraryProfiling = false;
  };
```

error:
```
error: builder for '/nix/store/znykhpsg57v38gdfjx0zwlh17716n6md-hashable-1.3.0.0.drv' failed with exit code 1;
       last 10 log lines:
       >   $, called at libraries/Cabal/Cabal/Distribution/Simple/Configure.hs:1024:20 in Cabal-3.2.1.0:Distribution.Simple.Configure
       >   configureFinalizedPackage, called at libraries/Cabal/Cabal/Distribution/Simple/Configure.hs:477:12 in Cabal-3.2.1.0:Distribution.Simple.Configure
       >   configure, called at libraries/Cabal/Cabal/Distribution/Simple.hs:625:20 in Cabal-3.2.1.0:Distribution.Simple
       >   confHook, called at libraries/Cabal/Cabal/Distribution/Simple/UserHooks.hs:65:5 in Cabal-3.2.1.0:Distribution.Simple.UserHooks
       >   configureAction, called at libraries/Cabal/Cabal/Distribution/Simple.hs:180:19 in Cabal-3.2.1.0:Distribution.Simple
       >   defaultMainHelper, called at libraries/Cabal/Cabal/Distribution/Simple.hs:116:27 in Cabal-3.2.1.0:Distribution.Simple
       >   defaultMain, called at Setup.hs:3:8 in main:Main
       > Setup: Encountered missing or private dependencies:
       > random >=1.0 && <1.2
```

**Comment by srid** [comment, marked as answer]
You are using [`defaults.settings.default`](https://flake.parts/options/haskell-flake.html#opt-perSystem.haskellProjects._name_.defaults.settings.default), right?

As the description says, ~~this~~ (the default value of this) only applies to local packages as well as packages imported via [project modules](https://haskell.flake.page/modules).

**Reply by srid** [reply]
> As the description says, this only applies to local packages as well as packages imported via [project modules](https://haskell.flake.page/modules).

Clarification: this semantics is for the *default* value of this option. The option itself is applied only for packages in the `packages` option. It doesn't apply it to the entire Haskell package set (that is not generally considered desirable anyway?).

**Reply by srid** [reply]
@abshekh What is the larger thing you are trying to achieve here?

**Reply by conscious-puppet** [reply]
yes the i was not using the flag correctly.
